homeassistant/components/geo_json_events/geo_location.py

```diff
@@ -59,63 +60,72 @@ def setup_platform(
+    manager = GeoJsonFeedEntityManager(
+        hass, async_add_entities, scan_interval, coordinates, url, radius_in_km
     )
+    await manager.async_init()
 
-    def start_feed_manager(event):
+    @callback
+    async def start_feed_manager(event=None):
         """Start feed manager."""
-        feed.startup()
+        await manager.async_update()
 
-    hass.bus.listen_once(EVENT_HOMEASSISTANT_START, start_feed_manager)
+    hass.bus.async_listen_once(EVENT_HOMEASSISTANT_START, start_feed_manager)
 
 
 class GeoJsonFeedEntityManager:
     """Feed Entity Manager for GeoJSON feeds."""
 
     def __init__(
-        self, hass, add_entities, scan_interval, coordinates, url, radius_in_km
+        self, hass, async_add_entities, scan_interval, coordinates, url, radius_in_km
     ):
         """Initialize the GeoJSON Feed Manager."""
 
         self._hass = hass
+        websession = aiohttp_client.async_get_clientsession(hass)
         self._feed_manager = GenericFeedManager(
+            websession,
             self._generate_entity,
             self._update_entity,
             self._remove_entity,
             coordinates,
             url,
             filter_radius=radius_in_km,
         )
-        self._add_entities = add_entities
+        self._async_add_entities = async_add_entities
         self._scan_interval = scan_interval
 
-    def startup(self):
-        """Start up this manager."""
-        self._feed_manager.update()
-        self._init_regular_updates()
+    async def async_init(self):
+        """Schedule initial and regular updates based on configured time interval."""
 
-    def _init_regular_updates(self):
-        """Schedule regular updates at the specified interval."""
-        track_time_interval(
-            self._hass, lambda now: self._feed_manager.update(), self._scan_interval
-        )
+        async def update(event_time):
+            """Update."""
+            await self.async_update()
+
+        # Trigger updates at regular intervals.
+        async_track_time_interval(self._hass, update, self._scan_interval)
+        _LOGGER.debug("Feed entity manager initialized")
+
+    async def async_update(self):
+        """Refresh data."""
+        await self._feed_manager.update()
+        _LOGGER.debug("Feed entity manager updated")
 
     def get_entry(self, external_id):
         """Get feed entry by external id."""
         return self._feed_manager.feed_entries.get(external_id)
 
-    def _generate_entity(self, external_id):
+    async def _generate_entity(self, external_id):
         """Generate new entity."""
         new_entity = GeoJsonLocationEvent(self, external_id)
         # Add new entities to HA.
-        self._add_entities([new_entity], True)
+        self._async_add_entities([new_entity], True)
 
-    def _update_entity(self, external_id):
+    async def _update_entity(self, external_id):
         """Update entity."""
         dispatcher_send(self._hass, f"geo_json_events_update_{external_id}")
```

Comment:
Please use the async dispatch interface when in async context.

Reply:
OK. Changed to async method in both occurrences.